File: array-hash/706-Design-Hashmap.go
type Pair struct {
	Key   int
	Value int
}

type MyHashMap struct {
	buckets    [][]Pair
	bucketSize int
}

func Constructor() MyHashMap {
	return MyHashMap{
		buckets:    make([][]Pair, 1000),
		bucketSize: 1000,
	}
}

func (this *MyHashMap) hash(key int) int {
	return key % this.bucketSize
}

func (this *MyHashMap) Put(key int, value int) {
	index := this.hash(key)
	bucket := this.buckets[index]
	for i, pair := range bucket {
		if pair.Key == key {
			bucket[i].Value = value
			return
		}
	}
	this.buckets[index] = append(bucket, Pair{Key: key, Value: value})
}

func (this *MyHashMap) Get(key int) int {
	index := this.hash(key)
	for _, pair := range this.buckets[index] {
		if pair.Key == key {
			return pair.Value
		}
	}
	return -1
}

func (this *MyHashMap) Remove(key int) {
	index := this.hash(key)
	bucket := this.buckets[index]
	for i, pair := range bucket {
		if pair.Key == key {
			this.buckets[index] = append(bucket[:i], bucket[i+1:]...)
			return
		}
	}
}

/**
* Your MyHashMap object will be instantiated and called as such:
* obj := Constructor();
* obj.Put(key,value);
* param_2 := obj.Get(key);
* obj.Remove(key);
 */

File: reference/array-hash/706-Design-Hashmap.py
class Pair:
    def __init__(self, key: int, value: int):
        self.key = key
        self.value = value

class MyHashMap:
    def __init__(self):
        self.bucket_size = 1000
        self.buckets = [[] for _ in range(self.bucket_size)]

    def hash(self, key: int) -> int:
        return key % self.bucket_size

    def put(self, key: int, value: int) -> None:
        index = self.hash(key)
        bucket = self.buckets[index]
        for pair in bucket:
            if pair.key == key:
                pair.value = value
                return
        bucket.append(Pair(key, value))

    def get(self, key: int) -> int:
        index = self.hash(key)
        bucket = self.buckets[index]
        for pair in bucket:
            if pair.key == key:
                return pair.value
        return -1

    def remove(self, key: int) -> None:
        index = self.hash(key)
        bucket = self.buckets[index]
        for i, pair in enumerate(bucket):
            if pair.key == key:
                bucket.pop(i)
                return
        


# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)